src/forms/redux-form/textInput.test.jsx

```diff
@@ -13,15 +14,47 @@ describe('redux-form TextInput', function() {
 			required: true
 		},
 		meta: {
-			error: 'Did you mean Batman and Robin?'
+			error: MOCK_ERROR
 		}
 	};
 
 	it('renders a TextInput component with expected attributes from mock data', () => {
 		const component = shallow(<ReduxFormTextInput {...formAttrs} />);
 
 		expect(component).toMatchSnapshot();
+	});
 
+	describe('validateAfterTouched', () => {
+		it('always renders the error when `validateAfterTouched` is not true', () => {
+			const component = shallow(<ReduxFormTextInput {...formAttrs} />);
+			expect(component.prop('error')).toBe(MOCK_ERROR);
+		});
+		describe('when true', () => {
+			it('does not render the error when touched is false', () => {
+				const props = {
```

Comment:
just pull these props out and set `props.meta.touched = true` or `false` in each test. I htink its more legible whats going on but you can disagree.